### DISCUSSION TITLE
Formatting data when querying with a join table

### DISCUSSION THREAD

**Discussion by sean-hill** [supabase/supabase#18990] [Questions] [answered]
Hey, I have a simple data structure:

### Products
* Part Number
### Locations
* Name
  
With a join table:

### Products_Locations
* product_id
* location_id
* quantity

This is my query:

```js
const { data, error } = await supabase
  .from('products')
  .select(`
    id, part_number,
    locations:products_locations(quantity, location:locations(id, name))
  `)
```

With my current response:

```json
[
  {
    "id": 10,
    "part_number": "123",
    "locations": [
      {
        "quantity": 5,
        "location": {
          "id": 1,
          "name": "a location"
        }
      }
    ]
  }
]
```

I would love to get rid of that duplicate nested `location` field and have a response like this:

```json
[
  {
    "id": 10,
    "part_number": "123",
    "locations": [
      {
        "id": 1,
        "quantity": 5,
        "name": "a location"
      }
    ]
  }
]
```

Essentially I'm trying to query all my products, and return a list of all the locations a product is apart of with how many products (`quantity`) are at each location.

Thanks!

**Comment by GaryAustin1** [comment, marked as answer]
You probably can use the spread operator to clean that up.
https://supabase.com/blog/postgrest-11-prerelease
See spreading related tables there.

**Reply by sean-hill** [reply]
@GaryAustin1 that was perfect!!! This was my solution:

```js
const { data, error } = await supabase
  .from('products')
  .select(
    `
    id,
    locations:products_locations(quantity, ...locations(id, name))
  `
  )
  .limit(20)
  .order('id', { ascending: false })
```

Which resulted in this response:

```json
[
  {
    "id": 10,
    "locations": [
      {
        "quantity": 5,
        "id": 1,
        "name": "J123"
      }
    ]
  }
]
```

Thanks so much!

**Reply by MateoWartelle** [reply]
Thanks
